.circleci/config.yml

```diff
@@ -1,75 +1,105 @@
-          python -m unittest discover parsers/test
-          npm install -g jsonlint
-          jsonlint -q config/*.json web/locales/*.json
+      - run:
+          name: Upgrade pip
+          command: sudo pip install -q --upgrade pip
+      - run:
+          command: |
+            set -eux -o pipefail
+            python --version
+            pip -v --version
+            sudo pip install --progress-bar=off -q flake8 pylint==2.3.1
+            sudo apt-get install libxml2-dev
+            sudo pip install --progress-bar=off -q -r parsers/requirements.txt
+            # stop the build if there are Python syntax errors or undefined names
+            flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
+            # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+            # flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+            pylint -E parsers/*.py -d unsubscriptable-object,unsupported-assignment-operation,unpacking-non-sequence
+            python -m unittest discover parsers/test
       - save_cache:
           key: pip-{{ .Branch }}-{{ checksum "parsers/requirements.txt" }}
           paths:
             - "~/.cache/pip"
-      - type: shell
-        name: Build
-        command: |
-          set -euo pipefail
-          docker-compose build
-          # Make sure files are available outside of container
-          CONTAINER_ID=$(docker create eu.gcr.io/tmrow-152415/electricitymap_web:latest)
-          docker cp $CONTAINER_ID:/home/web/public/dist web/public/dist
-        no_output_timeout: 30m
-      - type: shell
-        name: Start environment
-        command: |
-          set -euo pipefail
-          docker-compose up -d
-          sleep 20
-          docker-compose logs web
-      - type: shell
-        name: Run tests
-        command: |
-          docker-compose ps
-          curl --fail -i http://localhost:8000/
-          curl --fail -i http://localhost:8000/v1/co2?countryCode=FR
-          curl --fail -i http://localhost:8000/v1/exchanges?countryCode=FR
-          curl --fail -i http://localhost:8000/v1/production?countryCode=FR
-          curl --fail -i http://localhost:8000/v1/price?countryCode=FR
-          curl --fail -i http://localhost:8000/v2/history?countryCode=FR
+
+  lint_json:
+    docker:
+      - image: circleci/node:10.13
+    steps:
+      - checkout
+      - run:
+          command: |
+            set -eux -o pipefail
+            sudo npm install -g jsonlint
+            jsonlint -q config/*.json web/locales/*.json
+
+  lint_js:
+    docker:
+      - image: circleci/node:10.13
+    steps:
+      - checkout
+      - restore_cache:
+          key: node_modules-{{ .Branch }}-{{ checksum "web/yarn.lock" }}
+      - run:
+          command: |
+            set -eux -o pipefail
+            pushd web
```

Comment:
Can we remove this line?

Reply:
the next two commands require us to be in the "web" directory. 
I can also replace by a `cd` command?

Reply:
Ah ok.. Yeah, I think `cd` would be more standard if we're not using the dir stack, but it's a minor details :wink: